In [0]:
%pip install langchain langchain-huggingface langchain-community sentence-transformers
%pip install databricks-langchain
dbutils.library.restartPython()

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.5/2.5 MB 62.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 566.1/566.1 kB 31.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.0/1.0 MB 49.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 135.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 915.7/915.7 MB 32.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.2/12.2 MB 134.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 594.3/594.3 MB 42.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.2/10.2 MB 77.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 88.0/88.0 MB 89.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 954.8/954.8 kB 50.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 706.8/706.8 MB 34.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 193.1/193.1 MB 80.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 

In [0]:
%pip install itables
dbutils.library.restartPython()

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.4/2.4 MB 58.5 MB/s eta 0:00:00
Note: you may need to restart the kernel using %restart_python or dbutils.library.restartPython() to use updated packages.


In [0]:
from langchain_huggingface import HuggingFaceEmbeddings

# ============================================
# 1. GTE Model
# ============================================
gte_embeddings = HuggingFaceEmbeddings(
    model_name="thenlper/gte-large",
    model_kwargs={'device': 'cpu'},  # Use 'cuda' for GPU
    encode_kwargs={'normalize_embeddings': True}
)

# ============================================
# 2. BGE Model
# ============================================
bge_embeddings = HuggingFaceEmbeddings(
    model_name="BAAI/bge-large-en-v1.5",
    model_kwargs={'device': 'cpu'},
    encode_kwargs={'normalize_embeddings': True}
)

# ============================================
# 3. MiniLM Model
# ============================================
minilm_embeddings = HuggingFaceEmbeddings(
    model_name="sentence-transformers/all-MiniLM-L6-v2",
    model_kwargs={'device': 'cpu'},
    encode_kwargs={'normalize_embeddings': True}
)

print("✓ All embedding models loaded!")


modules.json:   0%|          | 0.00/385 [00:00<?, ?B/s]

README.md: 0.00B [00:00, ?B/s]

sentence_bert_config.json:   0%|          | 0.00/57.0 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/619 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/670M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/342 [00:00<?, ?B/s]

vocab.txt: 0.00B [00:00, ?B/s]

tokenizer.json: 0.00B [00:00, ?B/s]

special_tokens_map.json:   0%|          | 0.00/125 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/191 [00:00<?, ?B/s]

modules.json:   0%|          | 0.00/349 [00:00<?, ?B/s]

config_sentence_transformers.json:   0%|          | 0.00/124 [00:00<?, ?B/s]

README.md: 0.00B [00:00, ?B/s]

sentence_bert_config.json:   0%|          | 0.00/52.0 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/779 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.34G [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/366 [00:00<?, ?B/s]

vocab.txt: 0.00B [00:00, ?B/s]

tokenizer.json: 0.00B [00:00, ?B/s]

special_tokens_map.json:   0%|          | 0.00/125 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/191 [00:00<?, ?B/s]

modules.json:   0%|          | 0.00/349 [00:00<?, ?B/s]

config_sentence_transformers.json:   0%|          | 0.00/116 [00:00<?, ?B/s]

README.md: 0.00B [00:00, ?B/s]

sentence_bert_config.json:   0%|          | 0.00/53.0 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/612 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/90.9M [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/350 [00:00<?, ?B/s]

vocab.txt: 0.00B [00:00, ?B/s]

tokenizer.json: 0.00B [00:00, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/190 [00:00<?, ?B/s]

✓ All embedding models loaded!


In [0]:
# Sample documents to embed
documents = [
    "Machine learning is a subset of artificial intelligence",
    "Embeddings are numerical representations of text",
    "Vector databases store and search embeddings efficiently",
    "Deep learning uses neural networks with multiple layers",
    "Natural language processing enables computers to understand text",
    "Transformers revolutionized NLP with attention mechanisms",
    "Semantic search finds similar documents based on meaning",
    "Retrieval augmented generation combines search with generation",
]

# Create document IDs
doc_ids = list(range(1, len(documents) + 1))

In [0]:
import pandas as pd
from datetime import datetime

# Generate embeddings for all documents using each model
print("Generating GTE embeddings...")
gte_vectors = gte_embeddings.embed_documents(documents)

print("Generating BGE embeddings...")
bge_vectors = bge_embeddings.embed_documents(documents)

print("Generating MiniLM embeddings...")
minilm_vectors = minilm_embeddings.embed_documents(documents)

print("✓ All embeddings generated!")

# Print embedding dimensions
print(f"\nEmbedding Dimensions:")
print(f"  GTE: {len(gte_vectors[0])}")
print(f"  BGE: {len(bge_vectors[0])}")
print(f"  MiniLM: {len(minilm_vectors[0])}")


Generating GTE embeddings...
Generating BGE embeddings...
Generating MiniLM embeddings...
✓ All embeddings generated!

Embedding Dimensions:
  GTE: 1024
  BGE: 1024
  MiniLM: 384


In [0]:
from pyspark.sql.types import StructType, StructField, StringType, IntegerType, ArrayType, FloatType, TimestampType
from pyspark.sql.functions import current_timestamp, lit

# ============================================
# Option A: Unified Table with All Embeddings
# ============================================

# Create Pandas DataFrame with all embeddings
df_pandas = pd.DataFrame({
    'id': doc_ids,
    'text': documents,
    'gte_embedding': gte_vectors,
    'bge_embedding': bge_vectors,
    'minilm_embedding': minilm_vectors,
})

# Define schema for Spark DataFrame
schema = StructType([
    StructField("id", IntegerType(), True),
    StructField("text", StringType(), True),
    StructField("gte_embedding", ArrayType(FloatType()), True),
    StructField("bge_embedding", ArrayType(FloatType()), True),
    StructField("minilm_embedding", ArrayType(FloatType()), True),
])

# Convert to Spark DataFrame
df_spark = spark.createDataFrame(df_pandas, schema=schema)
df_spark = df_spark.withColumn("created_at", current_timestamp())

# Write to Delta table
df_spark.write \
    .mode("overwrite") \
    .format("delta") \
    .saveAsTable("users.anirvan_sen.embeddings_all_models")

print("✓ Unified embeddings table saved!")
display(spark.table("users.anirvan_sen.embeddings_all_models"))


✓ Unified embeddings table saved!


id text gte_embedding bge_embedding minilm_embedding created_at 1 Machine learning is a subset of artificial intelligence List(3.0898827E-4, 0.015768236, -0.015077546, -0.043820597, 0.0050207432, 0.004128571, -0.008725393, -0.001311115, 0.0094327275, 0.0507382, 0.009096327, 0.004829681, -0.006461243, -0.050563943, -0.011848135, -0.011544354, -0.031207299, -0.029132307, 0.008290318, -0.0063972687, -0.0011975683, 0.022208434, -0.0540464, -0.021469187, -0.03078138, 0.033175446, 0.0515184, -0.0017379643, 0.07657, 0.054323606, -0.0069384594, -0.05804896, 0.018908495, -0.053596344, -0.025838312, 3.3276633E-4, 0.027877642, -0.0041831266, -0.032311007, -0.062958635, 0.004180701, -0.005398196, 0.030487671, -0.057961114, -0.037662882, 0.013534322, -0.004739731, -0.039375946, 0.0048775687, -0.02672305, -0.004819178, 0.01160096, 0.021066513, -0.015909, 0.01512991, -0.0042793863, 0.018463802, -4.4805347E-4, -0.011533765, 0.021574158, 0.019433549, -0.017075818, 0.018465443, -0.041104052, 0.040909734, 0.00659755, -0.005448958, -0.010873007, 0.003801651, -0.044452507, -0.020766681, 0.013189367, -0.011979289, -0.027992882, -0.012914876, -0.013959088, 0.0048017823, 0.025289916, -0.019522311, 0.0025997732, 0.033010732, 0.055402182, -0.00566198, 0.04395545, -0.017843744, -0.06188489, 0.018120391, 0.030307349, -0.0054038805, -0.03624775, -0.011830259, 0.0374601, -0.004061541, 0.008552228, 0.028286934, 0.047256876, -0.030019619, 0.060448565, 0.01325707, 0.023520218, 0.058353256, 0.08280037, -0.010893495, 0.06401958, -0.054311205, 0.0033044422, 0.010974101, -0.023221366, 0.010471665, -0.045928642, 0.010101335, -0.01672447, 0.005067286, -0.010460455, -0.019949121, 0.046156503, -0.008348988, 0.040905967, -0.037966948, 0.017046059, -0.0066754227, 0.009198715, 0.03723871, 0.0038502975, 0.010799549, -0.00929098, -0.035326738, 0.04460476, -0.006207724, -0.0033250076, -0.010462402, -0.024638357, 0.0023405224, 0.02069977, 0.016791685, -0.023221314, -0.0075963982, 0.034027766, 0.016758706, -0.026132787, 0.044755086, 0.011500135, -0.019276539, 0.09285267, -0.012400916, 0.013003891, 0.0041803448, -0.008821944, 0.0023518598, 0.031039348, -0.044181477, 0.019900024, 3.916319E-5, 0.008024621, -0.02802707, -0.00886522, -0.023582231, 0.020761028, 0.027053842, 0.020959994, -0.014594929, -0.004467784, -0.005656867, 0.02557494, 0.001956508, 0.033176403, -0.06540137, -0.013434713, -0.029685747, -0.045267627, 0.0261952, -0.01430027, -0.0014702915, 0.015591496, 0.035861123, 0.07990512, 0.032783978, -0.0034707142, 0.03963117, 0.014562429, -0.026172832, -0.028239366, -0.007588951, 0.041318107, 0.014567054, 0.011375054, 0.005503124, -0.010216203, -0.043515082, -0.033570074, -0.0025637131, 0.047659215, -0.05145491, 0.015153511, -0.0035621063, 0.01861102, -0.027506582, 0.0042069554, -0.0059933513, -0.036308307, -0.046169303, 0.049651507, -0.013471812, 0.0036133514, -0.004571431, -0.009212293, 0.019007552, 0.05470474, -0.08447873, 0.039892122, 0.009233383, 0.0335765, 0.004541422, -0.035487235, 0.010928049, -0.006655718, -0.029123317, 0.02546444, -0.0071977144, -0.015427967, 0.012235364, -0.013524308, 0.034963362, 0.054275136, 0.011060381, -0.02547603, 0.006246412, 0.06161852, -0.027380971, 8.0775906E-4, 0.03339128, 0.041722924, 0.0058439705, 0.05148291, 0.038795736, 0.017909141, 0.03743039, 0.0530921, 0.0072851274, 0.020568285, 0.017278228, 0.0044188816, 0.038460657, 0.04946101, -0.016847994, 0.037600193, 0.0011688108, -0.01821368, -0.0043345215, 0.010531188, -0.014222025, 0.034470987, 0.03479201, 0.024289446, -0.04532293, 0.007683537, 0.03091755, 0.05698299, -0.019586138, -0.032905743, -0.03193768, 0.028767213, -0.023284992, -0.008032191, -0.011570889, 0.041800987, 0.023856532, 0.013433583, -0.013507423, -0.029065305, -0.057743784, -0.079844, -0.05694955, -0.04143569, -0.042212304, -0.013676327, 0.010394332, -0.048594356, 0.01924403, -0.0018860654, -0.0226521, -0.016743096, -0.012149714, 0.03521747, 0.018463887, 0.028443094, -0.060900543, 0.041994467, -0.025158279, 0.041587956


#### Now we have created three index  gte_embedding_vs_index , bge_embedding_vs_index , minilm_embedding_vs_index from catalog explorer ui

https://docs.databricks.com/aws/en/vector-search/create-vector-search#create-index-using-the-ui


### Evaluate Different embedding model

In [0]:
import mlflow
import pandas as pd
from mlflow.entities import Document
from mlflow.genai.scorers import RetrievalRelevance, RetrievalGroundedness
from databricks.vector_search.client import VectorSearchClient
from langchain_huggingface import HuggingFaceEmbeddings
from typing import List

# ============================================
# SETUP
# ============================================
client = VectorSearchClient()
username = spark.sql("SELECT current_user()").first()[0]
experiment_path = f"/Users/{username}/embedding_model_comparison_v3"
mlflow.set_experiment(experiment_path)

# ============================================
# DIVERSE EVALUATION QUERIES
# ============================================
eval_dataset = [
    {"inputs": {"query": "How do computers learn from data?"}},
    {"inputs": {"query": "Converting words into numbers"}},
    {"inputs": {"query": "Neural network architectures"}},
    {"inputs": {"query": "Text understanding by machines"}},
    {"inputs": {"query": "AI models that process sequences"}},
    {"inputs": {"query": "Finding similar content"}},
    {"inputs": {"query": "Attention mechanism in neural networks"}},
    {"inputs": {"query": "Dense vector representations"}},
    {"inputs": {"query": "Combining search results with language models"}},
    {"inputs": {"query": "Storing and querying high-dimensional data"}},
    {"inputs": {"query": "How to train a model on custom data?"}},
    {"inputs": {"query": "Difference between supervised and unsupervised learning"}},
    {"inputs": {"query": "vectors"}},
    {"inputs": {"query": "attention"}},
    {"inputs": {"query": "encoding text"}},
]

# ============================================
# INITIALIZE EMBEDDING MODELS
# ============================================
gte_model = HuggingFaceEmbeddings(model_name="thenlper/gte-large", model_kwargs={'device': 'cpu'}, encode_kwargs={'normalize_embeddings': True})
bge_model = HuggingFaceEmbeddings(model_name="BAAI/bge-large-en-v1.5", model_kwargs={'device': 'cpu'}, encode_kwargs={'normalize_embeddings': True})
minilm_model = HuggingFaceEmbeddings(model_name="sentence-transformers/all-MiniLM-L6-v2", model_kwargs={'device': 'cpu'}, encode_kwargs={'normalize_embeddings': True})

models = {
    "GTE": {"index_name": "users.anirvan_sen.gte_embedding_vs_index", "embedding_model": gte_model},
    "BGE": {"index_name": "users.anirvan_sen.bge_embedding_vs_index", "embedding_model": bge_model},
    "MiniLM": {"index_name": "users.anirvan_sen.minilm_embedding_vs_index", "embedding_model": minilm_model}
}

# ============================================
# CREATE RETRIEVER & RAG FACTORY
# ============================================
def create_retriever(index_name: str, model_name: str, embedding_model):
    @mlflow.trace(span_type="RETRIEVER")
    def retrieve(query: str) -> List[Document]:
        query_vector = embedding_model.embed_query(query)
        index = client.get_index(index_name=index_name)
        results = index.similarity_search(query_vector=query_vector, columns=["id", "text"], num_results=5)
        return [
            Document(id=str(row[0]), page_content=row[1], metadata={"model": model_name})
            for row in results.get('result', {}).get('data_array', [])
        ]
    return retrieve

def create_rag_app(retriever_fn):
    @mlflow.trace
    def rag_app(query: str) -> dict:
        docs = retriever_fn(query)
        context = "\n".join([doc.page_content for doc in docs])
        return {"response": f"Based on context: {context[:300]}..."}
    return rag_app

# ============================================
# RUN EVALUATIONS
# ============================================
print("Starting evaluations...\n")
for model_name, config in models.items():
    print(f"Evaluating {model_name}...")
    retriever = create_retriever(config["index_name"], model_name, config["embedding_model"])
    rag_app = create_rag_app(retriever)
    
    with mlflow.start_run(run_name=f"{model_name}_evaluation"):
        results = mlflow.genai.evaluate(
            data=eval_dataset,
            predict_fn=rag_app,
            scorers=[RetrievalRelevance(), RetrievalGroundedness()]
        )
        mlflow.log_param("embedding_model", model_name)
    
    print(f"✓ {model_name} done!\n")

# ============================================
# COMPARE RESULTS
# ============================================
print("\n" + "="*80)
print("MODEL COMPARISON RESULTS")
print("="*80 + "\n")

all_runs = mlflow.search_runs(experiment_names=[experiment_path])

filtered = all_runs[
    all_runs['metrics.retrieval_groundedness/mean'].notna()
    & all_runs['metrics.retrieval_relevance/mean'].notna()
    & all_runs['metrics.retrieval_relevance/precision/mean'].notna()
]


comparison_df = filtered[[
    'tags.mlflow.runName', 
    'params.embedding_model', 
    'metrics.retrieval_relevance/mean',
    'metrics.retrieval_groundedness/mean',
    'metrics.retrieval_relevance/precision/mean'
]].copy()

comparison_df.columns = ['Run Name', 'Model', 'Retrieval Relevance', 'Retrieval Groundedness', 'Precision']
comparison_df = comparison_df.round(4)

display(comparison_df)

# ============================================
# BEST PERFORMERS
# ============================================
print("\n" + "="*80)
print("BEST PERFORMERS")
print("="*80 + "\n")

best_relevance_idx = all_runs['metrics.retrieval_relevance/mean'].idxmax()
best_groundedness_idx = all_runs['metrics.retrieval_groundedness/mean'].idxmax()

best_relevance = all_runs.loc[best_relevance_idx]
best_groundedness = all_runs.loc[best_groundedness_idx]

print(f"🏆 Best Retrieval Relevance: {best_relevance['params.embedding_model']}")
print(f"   Score: {best_relevance['metrics.retrieval_relevance/mean']:.4f}\n")

print(f"🏆 Best Retrieval Groundedness: {best_groundedness['params.embedding_model']}")
print(f"   Score: {best_groundedness['metrics.retrieval_groundedness/mean']:.4f}\n")

# ============================================
# RANKING TABLE
# ============================================
print("\n" + "="*80)
print("RANKINGS")
print("="*80 + "\n")



ranking_df = filtered[['params.embedding_model', 'metrics.retrieval_relevance/mean', 'metrics.retrieval_groundedness/mean']].copy()
ranking_df.columns = ['Model', 'Retrieval Relevance', 'Retrieval Groundedness']

print("By Retrieval Relevance:")
display(ranking_df.sort_values('Retrieval Relevance', ascending=False).round(4).reset_index(drop=True))

print("\nBy Retrieval Groundedness:")
display(ranking_df.sort_values('Retrieval Groundedness', ascending=False).round(4).reset_index(drop=True))

print("\n✅ Evaluation Complete!")


[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.


2026/01/22 22:03:15 INFO mlflow.tracking.fluent: Experiment with name '/Users/sourav.banerjee@databricks.com/embedding_model_comparison_v3' does not exist. Creating a new experiment.


Starting evaluations...

Evaluating GTE...


2026/01/22 22:03:20 INFO mlflow.genai.utils.data_validation: Testing model prediction with the first sample in the dataset. To disable this check, set the MLFLOW_GENAI_EVAL_SKIP_TRACE_VALIDATION environment variable to True.


[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.


Evaluating:   0%|          | 0/15 [Elapsed: 00:00, Remaining: ?] 

[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. T

✓ GTE done!

Evaluating BGE...


2026/01/22 22:03:52 INFO mlflow.genai.utils.data_validation: Testing model prediction with the first sample in the dataset. To disable this check, set the MLFLOW_GENAI_EVAL_SKIP_TRACE_VALIDATION environment variable to True.


[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.


Evaluating:   0%|          | 0/15 [Elapsed: 00:00, Remaining: ?] 

[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. T

✓ BGE done!

Evaluating MiniLM...


2026/01/22 22:04:23 INFO mlflow.genai.utils.data_validation: Testing model prediction with the first sample in the dataset. To disable this check, set the MLFLOW_GENAI_EVAL_SKIP_TRACE_VALIDATION environment variable to True.


[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.


Evaluating:   0%|          | 0/15 [Elapsed: 00:00, Remaining: ?] 

[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. To disable this message, pass disable_notice=True.
[NOTICE] Using a notebook authentication token. Recommended for development only. For improved performance, please use Service Principal based authentication. T

✓ MiniLM done!


MODEL COMPARISON RESULTS



Run Name,Model,Retrieval Relevance,Retrieval Groundedness,Precision
MiniLM_evaluation,MiniLM,0.36,0.8,0.36
BGE_evaluation,BGE,0.3467,0.8667,0.3467
GTE_evaluation,GTE,0.36,0.8,0.36



BEST PERFORMERS

🏆 Best Retrieval Relevance: MiniLM
   Score: 0.3600

🏆 Best Retrieval Groundedness: BGE
   Score: 0.8667


RANKINGS

By Retrieval Relevance:


Model,Retrieval Relevance,Retrieval Groundedness
MiniLM,0.36,0.8
GTE,0.36,0.8
BGE,0.3467,0.8667



By Retrieval Groundedness:


Model,Retrieval Relevance,Retrieval Groundedness
BGE,0.3467,0.8667
MiniLM,0.36,0.8
GTE,0.36,0.8



✅ Evaluation Complete!


## Final Summary

* Retrieval Relevance (finding relevant documents): MiniLM and GTE are tied at 0.36 and are the best.
* Retrieval Groundedness (accurate/trustworthy responses): BGE is the best at 0.8667, while GTE and MiniLM are tied at 0.8.
* MiniLM and GTE are better at finding relevant documents, but BGE produces more accurate and trustworthy responses—choose based on whether you prioritize retrieval relevance or answer groundedness.